### PR TITLE
Add return type validator for SQF files

### PR DESCRIPTION
**Pull request by CreepPork**
**When merged this pull request will:**
- Add a python3 script that will check the file return type under these criteria:
  - If docblock indicated return type is `None`:
    - The last line character must be a `;`, `nil` or `};`.
  - If the indicated return type is something else from `None`:
    - It will emit an error with a possible fix.
- PR also fixes non-compliant files.

### CODE REVIEW

**Comment by CreepPork** [comment]
Current output:
```
➜  ZEN git:(return-type-checker) python3 ./tools/return_checker.py
Validating Return Types
-----------------------
Found 4 error(s)

ERROR: In file ./addons/context_actions/functions/fnc_canHealUnits.sqf
Incorrect return type, expected `None`
Found line    `_objects findIf {crew _x findIf {alive _x && {_x isKindOf "CAManBase"} && _fnc_filter} != -1} != -1`
Expected line `_objects findIf {crew _x findIf {alive _x && {_x isKindOf "CAManBase"} && _fnc_filter} != -1} != -1;`

ERROR: In file ./addons/context_menu/functions/fnc_compileActions.sqf
Incorrect return type, expected `None`
Found line    `*/`
Expected line `nil`

ERROR: In file ./addons/doors/functions/fnc_getState.sqf
Incorrect return type, expected `Door State <NUMBER>`
Found line    `[STATE_CLOSED, STATE_OPENED] select (_building animationSourcePhase ANIM_NAME_1(_door) > 0.5);`
Expected line `[STATE_CLOSED, STATE_OPENED] select (_building animationSourcePhase ANIM_NAME_1(_door) > 0.5)`

ERROR: In file ./addons/editor/functions/fnc_handleSearchClick.sqf
Incorrect return type, expected `Return Name <TYPE>`
Found line    `ctrlSetFocus _ctrlSearchCustom;`
Expected line `ctrlSetFocus _ctrlSearchCustom`

Return Validation FAILED
```

File: addons/editor/functions/fnc_handleSearchClick.sqf [hunk 1]
```diff
@@ -26,4 +26,4 @@ private _ctrlSearchEngine = _display displayCtrl IDC_RSCDISPLAYCURATOR_CREATE_SE
 _ctrlSearchEngine ctrlSetText "";
 
 _ctrlSearchCustom ctrlSetText "";
-ctrlSetFocus _ctrlSearchCustom;
+ctrlSetFocus _ctrlSearchCustom
```
Comment (mharis001) [inline addons/editor/functions/fnc_handleSearchClick.sqf]: Old was correct, the function header is just incomplete - return value should be "None".

Comment (mharis001) [inline addons/editor/functions/fnc_handleSearchClick.sqf]: Old was correct, the function header for this file is incomplete - return value should be "None".

**Comment by veteran29** [comment]
The script errors if file has 2 empty lines at the end:
```
Traceback (most recent call last):
  File "tools\return_checker.py", line 132, in <module>
    sys.exit(main())
  File "tools\return_checker.py", line 104, in main
    status = check_last_character(filepath, return_value)
  File "tools\return_checker.py", line 65, in check_last_character
    last_line_character = last_line[-1]
IndexError: string index out of range
```

**Pull request merged**